continuous_integration/gpuci/build.sh

```diff
@@ -40,6 +40,8 @@ gpuci_logger "Activate conda env"
 . /opt/conda/etc/profile.d/conda.sh
 conda activate dask
 
+mamba install -y 'aws-sdk-cpp<1.11'
```

Comment:
Is this due to some upstream issue? The actual change here looks totally fine, but it might be worth a comment with extra context

Reply:
Indeed, this is due to https://github.com/aws/aws-sdk-cpp/issues/2681, but the change here was just to test/confirm this. For now we will likely resolve the issue via https://github.com/rapidsai/cudf/pull/14173 and revert this change here.

Sorry for not commenting here earlier, I was trying to get this in before bringing people in to review the changes here. 🙂

Reply:
No worries. I just noticed that https://github.com/rapidsai/cudf/pull/14173 is merged -- does that mean we no longer need this here?

Reply:
It does now, we first had to fix some issues (mark old cuDF packages as broken and build new gpuCI docker images). I [reverted the `aws-sdk-cpp` changes](https://github.com/dask/distributed/pull/8201/commits/119dc1417ae2826e560096d8873aef1a5e82dd54) now, we just need to wait and see if everything passes and we should be good to merge this. 🙂